### DISCUSSION TITLE
Fail to integrate with Nuxt3

### DISCUSSION THREAD

**Discussion by maicWorkGithub** [vuejs/pinia#2137] [Help and Questions] [answered]
### Reproduction

https://stackblitz.com/edit/nuxt-starter-mjjyny?file=app.vue

### Steps to reproduce the bug

`npm run dev`

### Expected behavior

Should see the web render succeed.

### Actual behavior

```
pinia was called with no active Pinia. Did you forget to install pinia?
	const pinia = createPinia()
	app.use(pinia)
This will fail in production.
 ```

### Additional information

https://github.com/nuxt/nuxt/issues/20225

**Comment by posva** [comment, marked as answer]
You need to put the `useStore()` within the other store, not outside:

```ts
import { useBStore } from './b';
// cannot call useBStore() here
export const useAStore = defineStore('a', {
  state: () => {
    return {
      a: 22,
    };
  },
  actions: {
    increaseA() {
      this.a++;
// it must be here
      const bStore = useBStore();

      bStore.decrease();
    },
  },
});
```

**Reply by maicWorkGithub** [reply]
If so, if I have 10 places use bStore's method in aStore, I should write 10 times `const bStore = useBStore()`? And update bStore's state also rewrite `const bStore = useBStore()`？

**Reply by posva** [reply]
Yes, or use setup stores for these. There are quite a few examples in existing discussions and in docs: https://pinia.vuejs.org/cookbook/composing-stores.html

**Reply by maicWorkGithub** [reply]
Thanks, I thought I was using setup store already. 😂
